src/core/continuous_space.jl

```diff
@@ -352,16 +352,23 @@ The argument `method` provides three pairing scenarios
   agent is paired to its nearest neighbor. Similar to `:nearest`, each agent can belong
   to only one pair. This functionality is useful e.g. when you want some agents to be
   paired "guaranteed", even if some other agents might be nearest to each other.
+- `:types`: For mixed agent models only. Return every pair of agents within radius `r`
+  (similar to `:all`), with an additional constraint that excludes neighbors with the same
+  type. For example, a mixed agent model of `Union{Sheep,Wolf,Grass}`, pairs of 
+  `(Sheep,Sheep)` will be excluded. To also exclude `(Wolf,Grass)` pairs, provide a
+  scheduler that samples only matchable parings.

```

Comment:
In all honesty this all sounds very confusing for me, I think it needs to be documented simpler. 

What is "matchable pairings" ...?

(1) I don't think using `(Sheep, Sheep)` is a good idea. It specifies the thought process. It is not just `(sheep, Sheep)` pairs that are not allowed, it is also `(Wolf, Wolf)` and also `(Grass, Grass)`. What I recommend is to start the docstring with only a model `Union{Sheep, Wolf}` and say that only `(Sheep, Wolf)` pairs would be allowed.

(2) the second part of the docstring is unclear to me. Here we can say: In the case of having more agent types, e.g. `{Sheep, Wolf, Grass}`, and you want to _in general_ skip pairings that involve `Grass`, then simply provide a scheduler that doesn't schedule `Grass` types, i.e.: `scheduler = [a.id for a in allagents(model) of !(a isa Grass)]`.